recipes/imagl/all/conanfile.py

```diff
@@ -44,27 +61,70 @@ def source(self):
+
         minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("imagl requires C++20, which your compiler does not fully support.")
+        minimum_toolset_version = self._compiler_toolsets_minimum_version.get(str(self.settings.compiler), "0")
+        if not minimum_version:
+            self.output.warn("imaGL requires C++20. Your compiler is unknown. Assuming it supports C++20.")
+        elif lazy_lt_semver(compiler_version, minimum_version) or lazy_lt_semver(toolset_version, minimum_toolset_version):
+            raise ConanInvalidConfiguration("imaGL requires some C++20 features, which your {} {} compiler does not support.".format(str(self.settings.compiler), compiler_version)
+                                         + (" Your Visual Studio toolset version is {}; version {}+ is required.".format(toolset_version, str(minimum_toolset_version)) if minimum_toolset_version else ""))
+        elif str(self.settings.compiler) == "clang" and compiler_version == "11" and not self.options.allow_clang_11:
+            raise ConanInvalidConfiguration("Clang 11 is not currently supported by conan center index. To build imaGL, append '-o imagl:allow_clang_11=True --build missing' to your 'conan install' command line.")
         else:
-            self.output.warn("imagl requires C++20. Your compiler is unknown. Assuming it supports C++20.")
+            print("Your compiler is {} {} and is compatible.".format(str(self.settings.compiler), compiler_version))
+            if minimum_toolset_version != "0":
+                print("Its toolset is in version {} and is compatible. Minimum version is {}.".format(toolset_version, minimum_toolset_version))
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if not self._supports_jpeg:
+            del self.options.with_jpeg
+        if not str(self.settings.compiler) == "clang" or not str(self.settings.compiler.version) == "11":
+            del self.options.allow_clang_11
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
 
     def requirements(self):
         if self.options.with_png:
             self.requires("libpng/1.6.37")
+        if self._supports_jpeg and self.options.with_jpeg:
+            self.requires("libjpeg/9d")
 
     def _configure_cmake(self):
         if self._cmake:
             return self._cmake
-        self._cmake = CMake(self)
+        generator = "Ninja" if str(self.settings.compiler) == "Visual Studio" else None
+        self._cmake = CMake(self, generator=generator)
```

Comment:
Same remark as for the test package. Doesn't the default generator work?
```suggestion
        self._cmake = CMake(self)
```

Reply:
Same answer: no.